pylearn2/datasets/mnist_augmented.py

```diff
@@ -0,0 +1,80 @@
+"""
+Augmented MNIST wrapper class
+"""
+
+import os
+import numpy as np
+
+from pylearn2.datasets.dense_design_matrix import DenseDesignMatrix
+from pylearn2.scripts.dbm.augment_input import augment_input
+from pylearn2.utils import serial
+
+
+class MNIST_AUGMENTED(DenseDesignMatrix):
+
```

Comment:
In my private repo from grad school, I did this by just re-computing the features each time they're needed. It's only MNIST so on GPU this is not a big deal:
https://github.com/goodfeli/galatea/tree/master/dbm/inpaint/expdir

You could also write a function that takes a Dataset and returns an augmented Dataset (maybe a VectorSpacesDataset).

We can't have a separate class for every dataset and every possible processing of each dataset. That's just too much code to maintain.